machine/machine_core/ssh_connection.py

```diff
@@ -343,24 +343,14 @@ def upload(self, sources: Sequence[str], dest: str, relative_dir: str = TEST_DIR
         if not self.__ssh_direct_opt_var():
             self._ensure_ssh_master()
 
-        if use_scp:
```

Comment:
This is not so much "backwards" and more about supporting images/targets which don't have `rsync` installed. This might well still be the case on our fedora-coreos or other images, and it also breaks `--machine` cases. Does that hurt in any way? I'm very hesitant to drop this and turn rsync into a hard requirement.